### ISSUE TITLE
Support for JSON comments

### ISSUE THREAD

**Issue by brettz9** [atom/language-json#83]
## Summary

Support a JSON with comments mode so that users can opt not to get errors when comments are used within JSON files (or ideally, within specific JSON file names, like `.eslintrc` and `.eslintrc.json`).

## Motivation

It is annoying to see linting errors when viewing such a file, and it deters one from adding helpful comments to ESLint RC files (ESLint config files support this format: https://eslint.org/docs/user-guide/configuring/configuration-files#comments-in-configuration-files ).

## Describe alternatives you've considered

The existing `language-json-comments` package does not work for me.

While I understand JSON proper does not support comments, I think it makes sense for this existing package to implement since it appears that often the "json" extension is still used and Visual Studio treats this similarly as [another mode](https://code.visualstudio.com/docs/languages/json#_json-with-comments )

## Additional context

There is [a spec](https://github.com/komkom/jsonc/#syntax) for this and several parsers:

- [jsonc-parser](https://www.npmjs.com/package/jsonc-parser) by Microsoft
- [comment-json](https://www.npmjs.com/package/comment-json)
- [json-comments](https://www.npmjs.com/package/json-comments)
- [json-comment-parser](https://www.npmjs.com/package/json-comment-parser)
- [one-json](https://www.npmjs.com/package/one-json)

**Comment by lawrence-dol** [comment]
Yes, please! We use "human" JSON files for configuration extensively, and our parser supported standard curly-brace language comments, `/* ... */` and `//`. It would be awesome to have these rendered in the configured comment color.
